### DISCUSSION TITLE
Summary variables using DFN

### DISCUSSION THREAD

**Discussion by Guri080** [pybamm-team/PyBaMM#4327] [Q&A]
Hey!

I am trying to plot a discharge capacity vs cycle number graph. I was able to do that with the SPM model using the summary variables and matplotlib. However, I can not figure out how to use the DFN model's summary variables. When I try to simulate experiments using DFN it just says "At t = 439.545 and h = 2.83956e-16, the corrector convergence failed repeatedly or with |h| = hmin" and when I try to access the summary variables list it does not exist. I am pretty new to PyBaMM and batteries. I appreciate any help and insights!

**Comment by abhishek-appana** [comment]
Does the simulation complete or does it give an error or does it never end?

The summary variables can be plotted irrespective of whether using SPM or DFN. If the simulation results in an error, then the summary variables might not be available straight up!

If you are using casadi solver and getting an error, you can try adding this "`return_solution_if_failed_early=True` to the solver options and see if you can plot the variables.

You can also use `pybamm.set_logging_level("DEBUG")` to see exactly what is happening!

PS: I am pretty new as well to PyBaMM. Just giving some ideas as I saw the same thing happening in my code!

**Reply by Guri080** [reply]
Hello Abishek,

Apologise for the late reply and thanks for replying. The simulation did not result in an error and showed that the list does not exist when I tried to access the summary variables. Instead of using a pybamm.casadi solver I tried using pbyamm.simulation which gave the summary variables. However, the discharge capacity vs cycle number graph does not look correct, why do you think the difference is so big when I switch from spm to dfn? I will try to use the tips you mentioned! Thanks
